tools/projmgr/docs/Manual/Overview.md

```diff
@@ -0,0 +1,1317 @@
+
+- Unit/Integration Testing on simulation models (called Virtual Hardware) where Virtual Drivers implement the interface to simulated I/O.
+- Unit/Integration Testing the same software setup on a physical board where Hardware Drivers implement the interface to physical I/O.
+- System Testing where the software is combined with more software components that compose the final application.
+
+![Target and Build Types](./images/TargetBuild-Types.png "Target and Build Types")
+
+As the software may share a large set of common files, provisions are required to manage such projects.  The common way in other IDE's is to add:
+
+- **target-types** that select a target system. In the example this would be:
+  - `Virtual`: for Simulation Models.
+  - `Board`: for a physical evaluation board.
+  - `Production-HW`: for system integration test and the final product delivery.
+- **build-types** add the flexibility to configure each target build towards a specific testing.  It might be:
+  - `Debug`: for a full debug build of the software for interactive debug.
+  - `Test`: for a specific timing test using a test interface with code maximal optimization.
+  - `Release`: for the final code deployment to the systems.
+
+It is required to generate reproducible builds that can deployed on independent CI/CD test systems. To achieve that, the CMSIS Project Manager generates *.cprj output files with the following naming conventions:
+
+`<projectname>[.<build-type>][+target-type].cprj` this would generate for example: `Multi.Debug+Production-HW.cprj`
+
+This enables that each target and/or build type can be identified and independently generated which provides the support for test automation. It is however not required to build every possible combination, this should be under user control.
+
+**Flexible Builds: `Multi.cproject.yml`**
+
+```yml
+target-types:
+  - type: Board
+    board: NUCLEO-L552ZE-Q
+
+  - type: Production-HW
+    device: STM32L552XY         # specifies device
+
+  - type: Virtual
+    board: VHT-Corstone-300     # Virtual Hardware platform (appears as board)
+      
+build-types:
+  - type: Debug
+    optimize: debug
+    debug: on
+
+  - type: Test
+    optimize: max
+    debug: on
+
+  - type: Release
+    optimize: max
+    debug: off
+
+groups:
+  - group: My group1
+    files:
+      - file: file1a.c
+      - file: file1b.c
+      - file: file1c.c
+
+  - group: My group2
+      - file: file2a.c
+
+  - group: Test-Interface
+    for-type: .Test
+      - file: fileTa.c
+
+layers:
+  - layer: NUCLEO-L552ZE-Q/Board.clayer.yml   # tbd find a better way: i.e. $Board$.clayer.yml
+    for-type: +Board
+
+  - layer: Production.clayer.yml              # added for target type: Production-HW
+    for-type: +Production-HW
+
+  - layer: Corstone-300/Board.clayer.yml      # added for target type: VHT-Corstone-300
+    for-type: +VHT-Corstone-300
+
+components:
+  - component: Device:Startup
```

Comment:
Hmm would Device::Startup really be in a list of common components  e.g. would that apply as well to Virtual HW ?
I would have thought this would rather be part of the layers